### PR TITLE
feat: Provide a way to change radius style - MEED-2759 -Meeds-io/MIPs#81

### PR DESCRIPTION
**Pull request by SaraBoutej**
This PR add a new theme variable (borderRadius) to set the border radius in all the platform from the main setting app.

### CODE REVIEW

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/SiteBranding.vue [hunk 1]
```diff
@@ -251,6 +274,11 @@ export default {
         .catch(e => this.errorMessage = String(e))
         .finally(() => this.$root.loading = false);
     },
+    setBorderRadius(value) {
+      if (value >= 0) {
+        this.borderRadius = `${value}px`;
+      }
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/SiteBranding.vue]: Set all time the effective border radius to use, even `0px`

```suggestion
        this.borderRadius = `${value}px`;
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<template>
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue:21]: Missing License header

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    </v-btn>
+  </div>
+</template>
+<script>
+export default {
+  props: {
+    borderRadius: {
+      type: String,
+      default: null,
+    },
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: Use `v-model` with `this.$emit('input', this.borderRadius);` when the selected `borderRadius` changes. (use `watch` as well to reset value of `this.borderRadius = this.value`)

```suggestion
    value: {
      type: String,
      default: null,
    },
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      v-model="defaultBorderRadius"
+      :thumb-size="24"
+      tick-size="2"
+      thumb-label="always"
+      ticks
+      min="0"
+      max="20"
+      step="4"
+      color="primary"
+      @change="$emit('input', defaultBorderRadius)" />
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: No need when using `v-model`
```suggestion
/>
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/SiteBranding.vue (hunk 1 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/SiteBranding.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 2 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue:21]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 4 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 3 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 5]
```diff
@@ -0,0 +1,56 @@
+<template>
+  <div class="d-flex mt-8">
+    <v-btn 
+      text
+      small
+      @click="setBorderRadius(0)">
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]:
```suggestion
      @click="bordeRadius = 0">
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      thumb-label="always"
+      ticks
+      min="0"
+      max="20"
+      step="4"
+      color="primary" />
+    <v-btn
+      text
+      small
+      @click="setBorderRadius(20)">
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]:
```suggestion
      @click="bordeRadius = maxBorderRadius">
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      {{ $t('generalSettings.widgetAndAppStyle.square') }}
+    </v-btn>
+    <v-slider
+      v-model="bordeRadius"
+      :thumb-size="24"
+      tick-size="2"
+      thumb-label="always"
+      ticks
+      min="0"
+      max="20"
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]:
```suggestion
      :max="maxBorderRadius"
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  },
+  created() {
+    this.bordeRadius = this.value;
+  },
+  methods: {
+    setBorderRadius(value) {
+      this.bordeRadius = value;
+      this.$emit('input', this.bordeRadius);
+    }
+  }
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: Not needed anymore

```suggestion
```

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 5 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 6 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 7 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue (hunk 8 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/general-settings/components/branding/form/BorderRadiusSelector.vue]: removed

**Review by boubaker — APPROVED** [review]

**Pull request merged**